lib/checks/checks.go

```diff
@@ -157,29 +168,45 @@ func ValidateLocal(req LocalChecksRequest) (*LocalChecksResult, error) {
 		log.Infof("Skipping local checks due to %v set.", constants.PreflightChecksOffEnvVar)
 		return &LocalChecksResult{}, nil
 	}
+
 	err := req.CheckAndSetDefaults()
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+
 	stateDir, err := state.GetStateDir()
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	failedProbes, err := ValidateManifest(req.Manifest, req.Role, stateDir)
+
+	profile, err := req.Manifest.NodeProfiles.ByName(req.Role)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	failedProbes, err := ValidateManifest(req.Manifest, *profile, stateDir)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	failedProbes, err = ValidateDocker(req.Manifest.Docker(*profile), req.Docker, stateDir)
```

Comment:
This will override `failedProbes` above - don't you need to append them instead?

Reply:
This is gone as well.